js/interface.js

```diff
@@ -473,6 +473,29 @@ function initializeTable(options) {
   });
 }
 
+async function setTableData({ columns, rows, sortConfig }) {
+  await initializeTable({
+    columns,
+    rows: [],
+    initialLoad: true
+  });
+  
+  table = spreadsheet({
+    columns,
+    rows,
+    pageSize,
+    pageOffset,
+    sortConfig: sortConfig || getSortConfigForTable(getDataSourceQuery()),
+  });
+
+  updatePagination({
+    count: rows.length
+  });
+
+  const preparedData = table.prepareData(rows, columns);
+  HistoryStack.getCurrent().setData(preparedData);
```

Comment:
Why doesn't your linter add this automatically? There could be other errors in this PR that isn't easily detectable because the linter isn't working correctly.

Please investigate and fix your setup or the repository.

```suggestion

  HistoryStack.getCurrent().setData(preparedData);
```